server.js

```diff
@@ -0,0 +1,35 @@
+const express = require('express');
+const { createProxyMiddleware } = require('http-proxy-middleware');
+const port = 5000;
+
+const app = express();
+
+app.use('/comments', createProxyMiddleware({
+  target: 'http://localhost:4000/',
+  headers: {
+    method: 'GET'
+  },
+  changeOrigin: true
+}));
+
+app.use('/hashtags', createProxyMiddleware({
+  target: 'http://localhost:4001/',
+  headers: {
+    method: 'GET'
+  },
+  changeOrigin: true
+}));
+
+app.use('/users', createProxyMiddleware({
+  target: 'http://localhost:4002/',
+  headers: {
+    method: 'GET'
+  },
+  changeOrigin: true
+}));
+
```

Comment:
How this works is in the name, it's middleware. Which means that it consumes the Express decorated node http request and response objects, modifies them. All middleware for express does these things. In this case the modification is to the headers and that it re-routes (passes the request to) the target.

The modification with middleware could be anything tho. Want to add an emoji to every response body? middleware! Same flow.